KIPs/kip-81.md

```diff
@@ -0,0 +1,773 @@
+
+- Whenever its KLAY balance changes, V2 notifies the StakingTracker.
+- V2 stores a voter account address, and notifies the StakingTracker whenever it changes. The voter account can be changed by its admins.
+
+To support the new features, following functions are added.
+
+```solidity
+abstract contract CnStakingV2 {
+    event UpdateStakingTracker(address stakingTracker);
+    event UpdateVoterAddress(address voterAddress);
+
+    IStakingTracker public stakingTracker;
+    address public voterAddress;
+
+    /// @dev Set the initial stakingTracker address
+    /// Emits an UpdateStakingTracker event.
+    /// This function only works before contract initialization
+    function setStakingTracker(address _tracker) external beforeInit;
+
+    /// @dev Update the StakingTracker address this contract points to
+    function submitUpdateStakingTracker(address _tracker) external;
+    function updateStakingTracker(address _tracker) external onlyMultisigTx;
+
+    /// @dev Update the voter address of this GC
+    function submitUpdateVoterAddress(address _addr) external;
+    function updateVoterAddress(address _addr) external onlyMultisigTx;
+}
+```
+
+#### Notifying changes to StakingTracker
+
+To notify balance and voter account changes to StakingTracker contract, CnStakingV2 contract shall call the StakingTracker whenever after every change. For example,
+
+```solidity
+abstract contract CnStakingV2 {
+    function depositLockupStakingAndInit() payable beforeInit() {
+        // ...
+        stakingTracker.refreshStake(address(this));
+    }
+    function stakeKlay() payable {
+        // ...
+        stakingTracker.refreshStake(address(this));
+    }
+    function withdrawLockupStaking(address _to, uint256 _value) {
+        // ...
+        _to.transfer(_value);
+        stakingTracker.refreshStake(address(this));
+    }
+    function withdrawApprovedStaking(uint256 _approvedWithdrawalId) {
+        // ...
+        _to.transfer(_value);
+        stakingTracker.refreshStake(address(this));
+    }
+    function updateVoterAddress(address _addr) {
+        // ...
+        voterAddress = _addr;
+        stakingTracker.refreshVoter(address(this));
+    }
+}
+
+interface IStakingTracker {
+    /// @dev Re-evaluate Tracker contents related to the staking contract
+    /// @param staking  The CnStaking contract address
+    function refreshStake(address staking) external;
+
+    /// @dev Update a GC's voter address from given address
+    /// @param staking  The CnStaking contract address
+    function refreshVoter(address staking) external;
+}
+```
+
+#### Version identifier
+
+To distinguish the CnStakingContract and CnStakingV2 contracts, V2 will have VERSION value 2.
+
+```solidity
+abstract contract CnStakingV2 {
+    uint256 constant public VERSION = 2;
+}
+```
+
+### StakingTracker
+
+StakingTracker collects voting related data of each GC.
```

Comment:
```suggestion
StakingTracker collects voting related data of each GC member.
```

Governance Council member looks better to me instead of Governance Council. What do you think?